quinn-udp/src/windows.rs

```diff
@@ -1,23 +1,46 @@
 use std::{
     io::{self, IoSliceMut},
     mem,
+    net::{IpAddr, Ipv4Addr},
     os::windows::io::AsRawSocket,
+    ptr,
     sync::Mutex,
     time::Instant,
 };
 
+use once_cell::sync::Lazy;
 use windows_sys::Win32::Networking::WinSock;
 
-use super::{log_sendmsg_error, RecvMeta, Transmit, UdpSockRef, IO_ERROR_LOG_INTERVAL};
+use crate::{
+    cmsg::{self, CMsgHdr},
+    log_sendmsg_error, EcnCodepoint, RecvMeta, Transmit, UdpSockRef, IO_ERROR_LOG_INTERVAL,
+};
 
-/// QUIC-friendly UDP interface for Windows
+/// QUIC-friendly UDP socket for Windows
+///
+/// Unlike a standard Windows UDP socket, this allows ECN bits to be read and written.
 #[derive(Debug)]
 pub struct UdpSocketState {
     last_send_error: Mutex<Instant>,
 }
 
 impl UdpSocketState {
+    /// Create a QUIC-friendly UDP socket for Windows
+    ///
+    /// # Panics
+    /// - If `CMSG_LEN` is too small to store a [`WinSock::IN6_PKTINFO`] and one `i32` (for ECN)
+    /// - If [`WinSock::CMSGHDR`] has stricter alignment requirements than [`cmsg::Aligned`]
```

Comment:
`cmsg::Aligned` is an impl detail, so it's not useful to mention in the public docs. CI should catch such a case before anyone hits it anyway, so let's just cut the second bullet point here.